packages/jellyfish-api-core/src/category/poolpair.ts

```diff
@@ -153,10 +153,34 @@ export class PoolPair {
    * @param {string} metadata.to address of the owner of tokenTo
    * @param {string} metadata.tokenTo swap to token {symbol/id}
    * @param {number} [metadata.maxPrice] acceptable max price
+   * @param {'auto' | 'direct' | 'composite' | Array<string>} [path] swap path to use, defaults to auto. Provide array of poolpair IDs to set path manually.
+   * @param {boolean} [verbose=false] return pool path used, defaults to false
    * @return {Promise<string>} formatted as 'amount@token' swapped
    */
-  async testPoolSwap (metadata: PoolSwapMetadata): Promise<string> {
-    return await this.client.call('testpoolswap', [metadata], 'bignumber')
+  testPoolSwap (metadata: PoolSwapMetadata, path?: 'auto' | 'direct' | 'composite' | string[], verbose?: false): Promise<string>
+
+  /**
+   * Create a test pool swap transaction to check pool swap's return result
+   *
+   * @param {PoolSwapMetadata} metadata a provided information to create test pool swap transaction
+   * @param {string} metadata.from address of the owner of tokenFrom
+   * @param {string} metadata.tokenFrom swap from token {symbol/id}
+   * @param {number} metadata.amountFrom amount from tokenA
+   * @param {string} metadata.to address of the owner of tokenTo
+   * @param {string} metadata.tokenTo swap to token {symbol/id}
+   * @param {number} [metadata.maxPrice] acceptable max price
+   * @param {'auto' | 'direct' | 'composite' | Array<string>} [path] swap path to use, defaults to auto. Provide array of poolpair IDs to set path manually.
+   * @param {boolean} [verbose=true] return pool path used, defaults to false
+   * @return {Promise<string>} formatted as 'amount@token' swapped
+   */
+  testPoolSwap (metadata: PoolSwapMetadata, path?: 'auto' | 'direct' | 'composite' | string[], verbose?: true): Promise<TestPoolSwapVerboseResult>
+
+  async testPoolSwap (
+    metadata: PoolSwapMetadata,
+    path: 'auto' | 'direct' | 'composite' | string[] = 'auto',
```

Comment:
Could we have all the conditions tested?

Reply:
And what is `string[]`?

Reply:
`string[]` is an array of poolpair IDs, for when the user wants to manually specify the composite swap path.